File: backend/hospital-service/src/main/java/com/example/hospitalservice/model/Hospital.java
package com.example.hospitalservice.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Entity
@Table(name = "hospitals")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Hospital {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String address;
    private String phone;
    private String email;

    private double latitude;
    private double longitude;

    @ElementCollection
    @CollectionTable(name = "hospital_specialties", joinColumns = @JoinColumn(name = "hospital_id"))
    @Column(name = "specialty")
    private List<String> specialties;
}

File: mobile/app/src/main/java/com/example/ambulanceapp/beans/Hospital.kt
package com.example.ambulanceapp.beans

import org.simpleframework.xml.Element
import org.simpleframework.xml.ElementList
import org.simpleframework.xml.Root

@Root(name = "hospital")
data class Hospital(
    @field:Element(name = "id")
    var id: Long = 0,

    @field:Element(name = "name")
    var name: String? = null,

    @field:Element(name = "address")
    var address: String? = null,

    @field:Element(name = "phone")
    var phone: String? = null,

    @field:Element(name = "email")
    var email: String? = null,

    @field:Element(name = "latitude")
    var latitude: Double = 0.0,

    @field:Element(name = "longitude")
    var longitude: Double = 0.0,
    @field:ElementList(entry = "specialty", inline = true, required = false)
    var specialties: List<String>? = null
)